runtime/astar/src/xcm_config.rs

```diff
@@ -94,8 +94,23 @@ pub type FungiblesTransactor = FungiblesAdapter<
     CheckingAccount,
 >;
 
+pub type PSP34Transactor = PSP34MutateAdapter<
+    // Use this non-fungibles implementation:
+    Contracts,
```

Comment:
That's a fine approach but you could make it simpler (and less fragmented) by just implementing a new _adapter_ which is trait bound to `pallet_contracts::Config` and have it part of this PR (instead of forking substrate). Realistically, if we were doing this (or when we do it), it would be like this since those changes aren't exactly relevant for `pallet-contracts`.